### ISSUE TITLE
HTTPS doesn't work, and quite incomplete instructions

### ISSUE THREAD

**Issue by dchmelik** [teeli/urltitle#19]
I have the below; verified with './configure' and finding paths & dates on system (of my eggdrop binary, TCL) which not all people who use eggdrops know how to do that or about TCL & TLS (in fact most eggdrop users I know don't know.)  I'd have appreciated a chance or such instructions to find it out last time (could be added to a page or part of your main open issue, 'Broken URLs') and instructions if one needs to do anything in eggdrop.conf (necessary in your script's description section.) 

- eggdrop 1.8.4
- Tcl version: 8.3.5
- SSL/TLS Support: yes (OpenSSL 1.0.2s  28 May 2019)
- urltitle 0.11

The problem is HTTPS URLs never work (most/all HTTP ones work.)  I think you were right last time it's a configuration problem, because I solved it once or twice last few years maybe somehow changing that.  I just forgot and don't understand, for example, most the eggdrop.conf SSL section (didn't even recall it's related to TLS)... neither do most people I know who use eggdrops... so we need to know what to set in eggdrop.conf, not to mention different scripts use different parts of that and many/most people will have no idea which yours uses.  Hundreds HTTPS URLs haven't been working over several weeks, and of course, I can't go back to get all, but below are examples of some main ones that when are pasted as their own line on IRC, nothing happens (only Google displays at HTTP.)

https://amazon.com
https://apple.com
https://facebook.com
https://google.com
https://microsoft.com
https://twitter.com
https://yahoo.com

**Comment by ramsesatabusimbel** [comment]
You need some packages for uhm tcl I think. Try tcl-tls and what not. Did a fresh setup of eggdrop not long ago, tho already forgot what was needed for https links.

**Comment by dchmelik** [comment]
> You need some packages for uhm tcl I think. Try tcl-tls and what not. [...]

That's working because I have a current stable eggdrop botnet that requires TCL-TLS for netbots to work, and I currently use netbots commands probably weekly/monthly.

**Comment by ramsesatabusimbel** [comment]
What is the OS? Do you have the tclcurl installed? And does it have all repository's setup..

**Comment by teeli** [comment]
Do you get any errors in eggdrop logs when trying to fetch HTTPS URLs?

**Comment by ramsesatabusimbel** [comment]
I dont all urls show. That being linked that is.

**Comment by dchmelik** [comment]
> I dont all urls show. That being linked that is.

It already sounded like that on your first reply, but this is my issue...

I have TCLTLS working because the bot uses it to communicate with other bots.  If I recall correctly there are some TCLTLS/SSL requirements--undocumented--needed to get HTTPS working.  For some reason it's working now (I may have done some updates, maybe with help) but I'm sure at any next installation, I will have forgotten, and anyone not yet an eggdrop & TCLTLS/SSL advanced expert won't know either, so there's that part of this issue... 

I don't know what repositories have to do with this.

**Comment by ramsesatabusimbel** [comment]
I think I remember me doing the easy way and just installed with apt build-dep eggdrop. Are you using tdom, because for some reason i remember one of the packages gave https urls showing by installing additional packages. And perhaps tclCurl.. Might allready be installed with botnets requered packages.

**Comment by dchmelik** [comment]
I avoid such (apt, newer dependency-based) package managers.  I don't know if I have those libraries, but as said, it works now.

**Comment by ramsesatabusimbel** [comment]
Yeah I usually do too, limited space on vps and all that. What package/packages gave the show of https then? Would be great if you named them, for the future?

**Comment by dchmelik** [comment]
> [...] What package/packages gave the show of https then? [...]

Please restate?

**Comment by ramsesatabusimbel** [comment]
What did you install that made https links to show?

**Comment by dchmelik** [comment]
> What did you install that made https links to show?

This script?  I don't know dependencies (part of more strictly Unix[-like] OSes I I use... at least on most *BSD remote shells... in case of GNU/Linux aren't Debian or newer) nor eggdrop.conf settings... those various things are what _I_ asked to be documented...!  I've almost lost count of the number of times since 2014 or earlier I setup eggdrops and have no idea what settings are needed (still don't!)
